src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20230808_203400.java

```diff
@@ -57,28 +58,39 @@ public class MigrationEntry20230808_203400 extends MigrationEntry {
 
     private final Optional<CIVCSMigrationService> ciMigrationService;
 
-    private final AbstractVersionControlService versionControlService;
+    private final Optional<AbstractVersionControlService> versionControlService;
+
+    private final Environment environment;
 
     private final UrlService urlService = new UrlService();
 
     private final CopyOnWriteArrayList<ProgrammingExerciseParticipation> errorList = new CopyOnWriteArrayList<>();
 
+    private static final List<String> PROGRAMMING_EXERCISE_RELATED_PROFILES = List.of("bamboo", "bitbucket", "gitlab", "jenkins", "gitlabci", "localvc", "localci");
+
     public MigrationEntry20230808_203400(ProgrammingExerciseRepository programmingExerciseRepository,
             SolutionProgrammingExerciseParticipationRepository solutionProgrammingExerciseParticipationRepository,
             TemplateProgrammingExerciseParticipationRepository templateProgrammingExerciseParticipationRepository,
             ProgrammingExerciseStudentParticipationRepository programmingExerciseStudentParticipationRepository, AuxiliaryRepositoryRepository auxiliaryRepositoryRepository,
-            Optional<CIVCSMigrationService> ciMigrationService, AbstractVersionControlService versionControlService) {
+            Optional<CIVCSMigrationService> ciMigrationService, Optional<AbstractVersionControlService> versionControlService, Environment environment) {
         this.programmingExerciseRepository = programmingExerciseRepository;
         this.solutionProgrammingExerciseParticipationRepository = solutionProgrammingExerciseParticipationRepository;
         this.templateProgrammingExerciseParticipationRepository = templateProgrammingExerciseParticipationRepository;
         this.programmingExerciseStudentParticipationRepository = programmingExerciseStudentParticipationRepository;
         this.auxiliaryRepositoryRepository = auxiliaryRepositoryRepository;
         this.ciMigrationService = ciMigrationService;
         this.versionControlService = versionControlService;
+        this.environment = environment;
     }
 
     @Override
     public void execute() {
+        List<String> activeProfiles = List.of(environment.getActiveProfiles());
+        if (activeProfiles.stream().anyMatch(PROGRAMMING_EXERCISE_RELATED_PROFILES::contains)) {
```

Comment:
```suggestion
        if (activeProfiles.stream().noneMatch(PROGRAMMING_EXERCISE_RELATED_PROFILES::contains)) {
```
if (none of the active profiles are programming related) {
   return;
}

Inverted logic?